operator/v1/types_dns.go

```diff
@@ -0,0 +1,98 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// DNS manages CoreDNS component to provide a name resolution service
+// in the cluster.
+type DNS struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the specification of the desired behavior of the DNS.
+	Spec DNSSpec `json:"spec,omitempty"`
+	// status is the most recently observed status of the DNS.
+	Status DNSStatus `json:"status,omitempty"`
+}
+
+// DNSSpec is the specification of the desired behavior of the DNS.
+type DNSSpec struct {
+}
+
+// DNSStatus defines the observed status of the DNS.
+type DNSStatus struct {
+	// ClusterIP is the internal service IP reserved for cluster DNS service.
+	//
+	// Pods can use this IP to resolve domain names.
+	// This can be facilitated by one of these ways:
+	// * Enabling kubelet to inject this service IP in pod's nameserver list.
+	//   (this could be /etc/resolv.conf in some of the linux distributions)
+	// * Specify this service IP in pod's spec.dnsConfig.nameservers list.
+	// * Explicitly use this service IP as resolver.
+	//   Example: dig foo.com @<service IP>
+	//
+	// More info: https://kubernetes.io/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies
+	ClusterIP string `json:"clusterIP"`
+
+	// ClusterDomain is the local cluster DNS domain suffix for DNS services.
+	// Use domain names as defined in RFC 1034 (A-Z, a-z, 0-9 and the hyphen)
```

Comment:
Since this is in status, perhaps "Use domain names [...]" should be "This will be a domain name [...]".

Reply:
fixed